aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,296 @@
+
+Host Configuration Values
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Application Name ``string``
+    Key: ``applicationName``. This configuration setting specifies the value that will be returned from ``IHostingEnvironment.ApplicationName``.
+
+Capture Startup Errors ``bool``
+    Key: ``captureStartupErrors``. Defaults to ``false``. When ``false``, errors during startup result in the host exiting. When ``true``, the host will capture any exceptions from the ``Startup`` class, and attempt to start the server anyway, displaying an error page (generic, or detailed, based on the Detailed Errors setting, below) in response to every request. Set using the ``CaptureStartupErrors`` method.
+
+.. code-block:: c#
+
+    .CaptureStartupErrors(true)
+
+Content Root ``string``
+    Key: ``contentRoot``. Defaults to the folder where the application assembly resides (for Kestrel; IIS will use the web project root by default). This setting determines where ASP.NET Core will begin searching for content files, such as MVC Views. Also used as the base path for the Web Root setting, below. Set using the ``UseContentRoot`` method. Path must exist, or host will fail to start.
+
+.. code-block:: c#
+
+    .UseContentRoot("c:\\mywebsite")
+
+Detailed Errors ``bool``
+    Key: ``detailedErrors``. Defaults to ``false``. When ``true`` (or when Environment is set to "Development"), the app will display details of startup exceptions, instead of just a generic error page. Set using ``UseSetting``.
+
+.. code-block:: c#
+
+    .UseSetting("detailedErrors", "true")
+
+When Detailed Errors is set to ``false`` and Capture Startup Errors is ``true``, a generic error page is displayed in response to every request to the server.
+
+.. image:: hosting/_static/generic-error-page.png
+
+When Detailed Errors is set to ``true`` and Capture Startup Errors is ``true``, a detailed error page is displayed in response to every request to the server.
+
+.. image:: hosting/_static/detailed-error-page.png
+
+Environment ``string``
+    Key: ``environment``. Defaults to "Production". May be set to any value. Predefined values include "Development" and "Staging" in addition to "Production". Values are not case sensitive. See :doc:`environments`. Set using the ``UseEnvironment`` method.
+
+.. code-block:: c#
+
+    .UseEnvironment("Development")
+
+.. note:: By default, the environment is read from the ``ASPNETCORE_ENVIRONMENT`` environment variable. When using Visual Studio, the environment will be based on settings in the *launchSettings.json* file.
+
+Server URLs ``string``
+    Key: ``urls``. Set to a semicolon (;) separated list of URL prefixes to which the server should response. For example, "http://localhost:123". The domain/host name can be replaced with "*" to indicate the server listen to any request coming to any IP address or host on the specified port (for example, "http://*:5000"). The protocol ("http://" or "https://") must be included with each URL.
```

Comment:
I don't think so, because we don't have https by default, however it does need adjusting a little;

> The domain/host name can be replaced with "\*" to indicate the server should listen to requests on any IP address or host using the specified port _and protocol_ (for example, "http://\*:5000" *or "https://*\**:5001"*).